src/App.tsx

```diff
@@ -1,35 +1,51 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
-
-  // this code stops the timer
-  clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="clock">
-        <strong>{clockName}</strong>
-        {' time is '}
-        {date.toLocaleTimeString()}
-      </div>
-    </div>
-  );
+type State = {
+  clockName: string,
+  hasClock: boolean;
 };
 
-export default App;
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: getRandomName(),
+    hasClock: true,
+  };
+
+  nameTimer = 0;
+
+  componentDidMount() {
+    this.nameTimer = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+
+      this.nameTimer = window.setInterval(() => {
+        this.setState({ clockName: getRandomName() });
+      }, 3300);
+    });
+
+    document.addEventListener('contextmenu', () => {
+      this.setState({ hasClock: false });
+      clearInterval(this.nameTimer);
+    });
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {this.state.hasClock && <Clock clockName={this.state.clockName} />}
```

Comment:
you can move clockName logic into the Clock component and do not pass it in props
```suggestion
        {this.state.hasClock && <Clock clockName={this.state.clockName} />}
```